src/Entity/Appointment.php

```diff
@@ -42,28 +43,36 @@ class Appointment
     /**
      * @ORM\Column(type="datetime", nullable=true)
      */
-    private ?DateTime $dateTime;
+    private ?DateTimeInterface $dateTime;
 
     /**
      * @ORM\OneToMany(targetEntity="App\Entity\Talk", mappedBy="appointment")
      */
-    private TalkCollection $talks;
+    private Collection $talks;
 
-    public function __construct(string $title, ?string $text = null, ?DateTime $dateTime = null, array $talks = [])
+    public function __construct(string $title, ?string $text = null, ?DateTimeInterface $dateTime = null)
     {
         $this->title = $title;
         $this->text = $text;
         $this->dateTime = $dateTime;
-        $this->talks = new TalkCollection(...$this->talks);
+        $this->talks = new ArrayCollection();
     }
 
     public function getTitle(): string
     {
         return $this->title;
     }
 
-    public function getDateTime(): ?DateTime
+    public function getDateTime(): ?DateTimeInterface
     {
         return $this->dateTime;
     }
+
+    public function addTalk(Talk $talk): self
```

Comment:
Ich bin absolut kein Fan von fluent settern usw., da ich es nicht logisch finde, dass eine Methode einen Rückgabewert hat, der am Namen und der Aufgabe der Methode nicht erkennbar ist und ausschließlich dem Schreibkomfort dient.

Ist jetzt aber kein Weltuntergang. Wenn es dich glücklich macht, dann machen wir es so ;). Wäre nur gut, wenn wir uns da einmal global fürs Projekt festlegen.

Reply:
Ich bin für eine Einigung. Wir können auch gerne Immutable Objects nutzen und `hassers/withers` bauen, dann ist klar, warum diese einen Rückgabetyp `self` haben, weil sie eben ein neues, mutiertes Objekt zurückgeben...

Beispiel:
```php
public function withTalk(Talk $talk): self
{
  $new = clone $this;
  $new->talks[] = $talk;
  return $new;
}
```
Hab mich einfach nur daran orientiert, was das Maker Bundle ausgespuckt hat und da sind die Setter alle fluent.

Reply:
Klingt interessant. Muss ich mir noch anschauen, da ich noch nie so gearbeitet habe. Würde Doctrine nicht beispielsweise die alte Entity behalten statt des Klons? Oder würde man nicht irgendwann den Speicher voller ungenutzter Objekte haben, da ich ja immer wieder klone?

Reply:
Ich hab dir in eb076ac mal ein Beispiel eingefügt.

Reply:
Also ich vote dann für nicht-fluent Setters usw. 😇